File: WebAPI/WebAPI/Models/SubCategory.cs
namespace WebAPI.Models;

public class SubCategory
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    public Category Category { get; set; }
    public Guid CategoryID { get; set; }
    
    public AppUser Author { get; set; }
    public Guid AuthorId { get; set; }
    public List<Article> Articles { get; set; }
}

File: client/src/Pages/Articles/Components/Sidebar/SubCategory/SubCategory.js
import { useState } from 'react'
import { useParams } from 'react-router-dom'

import ArtilceLink from '../ArticleLink/ArtilceLink'
import getSubCategoryState from './Utility/getSubCategoryState'
import { IoMdArrowDropright, IoMdArrowDropdown } from 'react-icons/io' 

export default function SubCategory({subcategory, linkTo}) {
    const { id } = useParams()
    const [isOpen, setIsOpen] = useState(getSubCategoryState(subcategory,id))

    
    if(!isOpen){
        return (
            <li className='category-li' onClick={()=>setIsOpen(true)}>
                <IoMdArrowDropright />
                {subcategory.name}
            </li>
        )
    }

    return (
        <>
            <li className='category-li active' onClick={()=>setIsOpen(false)}>
                <IoMdArrowDropdown />
                {subcategory.name}
            </li>
            <ul className='article-list'>
                <ArtilceLink articles={subcategory.articles} linkTo={linkTo}/>
            </ul>
        </>
        
    )
}
